osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/PushSetupAlarmConverter.java

```diff
@@ -65,11 +96,38 @@ public PushSetupAlarm convertFrom(
     }
     final PushSetupAlarm.Builder builder = new PushSetupAlarm.Builder();
 
-    final String destination = source.getHost() + ":" + source.getPort();
-    final SendDestinationAndMethod sendDestinationAndMethod =
-        new SendDestinationAndMethod(
-            TransportServiceType.TCP, destination, MessageType.MANUFACTURER_SPECIFIC);
-    builder.withSendDestinationAndMethod(sendDestinationAndMethod);
+    if (source.getHost() != null && source.getPort() != null) {
```

Comment:
You already know if host and port is available. And you can retrieve them from pushSetupAlarm instead of source.

Why not including line 100 - 104 after line 59?

Reply:
It's in two different functions. This is `convertFrom`, line 59 is in `convertTo`